iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BatteryAdderImpl.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.impl;
+
+import com.powsybl.iidm.network.BatteryAdder;
+
+/**
+ * @author Ghiles Abdellah <ghiles.abdellah at rte-france.com>
+ */
+public class BatteryAdderImpl extends AbstractInjectionAdder<BatteryAdderImpl> implements BatteryAdder {
+
+    private final VoltageLevelExt voltageLevel;
+
+    private double p0 = Double.NaN;
+
+    private double q0 = Double.NaN;
+
+    private double minP = Double.NaN;
+
+    private double maxP = Double.NaN;
+
+    public BatteryAdderImpl(VoltageLevelExt voltageLevel) {
+        this.voltageLevel = voltageLevel;
```

Comment:
Assert that voltageLevel is not Null using objects.requireNonNull

Reply:
Done